bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/service/TriggerLocationCompactService.java

```diff
@@ -0,0 +1,102 @@
+import java.util.Map;
+import org.apache.bookkeeper.common.util.JsonUtil;
+import org.apache.bookkeeper.http.HttpServer;
+import org.apache.bookkeeper.http.service.HttpEndpointService;
+import org.apache.bookkeeper.http.service.HttpServiceRequest;
+import org.apache.bookkeeper.http.service.HttpServiceResponse;
+import org.apache.bookkeeper.proto.BookieServer;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * HttpEndpointService that handle force trigger entry location compact requests.
+ *
+ * <p>The PUT method will trigger entry location compact on current bookie.
+ *
+ * <p>The GET method will get the entry location compact running or not.
+ * Output would be like:
+ *        {
+ *           "is_entry_location_compact" : "false"
+ *        }
+ */
+
+public class TriggerLocationCompactService implements HttpEndpointService {
+
+    static final Logger LOG = LoggerFactory.getLogger(TriggerLocationCompactService.class);
+
+    private final BookieServer bookieServer;
+
+    public TriggerLocationCompactService(BookieServer bookieServer) {
+        this.bookieServer = checkNotNull(bookieServer);
+    }
+
+    @Override
+    public HttpServiceResponse handle(HttpServiceRequest request) throws Exception {
+        HttpServiceResponse response = new HttpServiceResponse();
+
+        if (HttpServer.Method.PUT.equals(request.getMethod())) {
+            String requestBody = request.getBody();
+            String output = "Not trigger Entry Location RocksDB compact.";
+            if (StringUtils.isNotBlank(requestBody)) {
+                @SuppressWarnings("unchecked")
+                Map<String, Object> configMap = JsonUtil.fromJson(requestBody, HashMap.class);
+                Boolean isEntryLocationCompact = (Boolean) configMap.getOrDefault("entryLocationRocksDBCompact", false);
+                boolean isCompactRunning = bookieServer.getBookie().getLedgerStorage().isEntryLocationCompact();
+                if (isEntryLocationCompact && !isCompactRunning){
+                    bookieServer.getBookie().getLedgerStorage().entryLocationCompact();
+                    output = "Triggered Entry Location RocksDB compact on bookie: " + bookieServer.getBookieId();
+                } else if (isCompactRunning) {
+                    output = "Entry Location RocksDB compact already running on bookie: " + bookieServer.getBookieId();
+                }
+            }
+            String jsonResponse = JsonUtil.toJson(output);
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("output body:" + jsonResponse);
+            }
+            response.setBody(jsonResponse);
+            response.setCode(HttpServer.StatusCode.OK);
+            return response;
+        } else if (HttpServer.Method.GET == request.getMethod()) {
+            boolean isCompactRunning = bookieServer.getBookie().getLedgerStorage().isEntryLocationCompact();
```

Comment:
We'd better show the detailed compacting info for each ledger directory instead of the total one

Reply:
Now return all entry location rocksDB compact status.